src/debugging/netSdk/NetSdkDebugHelper.ts

```diff
@@ -38,14 +46,70 @@ export class NetSdkDebugHelper extends NetCoreDebugHelper {
         return configurations;
     }
 
+    public async afterResolveDebugConfiguration(context: DockerDebugContext, debugConfiguration: DockerDebugConfiguration): Promise<void> {
+        const runTasks: Task[] = await tasks.fetchTasks({ type: NetSdkRunTaskType }) || [];
+        const runTaskFirstEntry = runTasks.find(t => t.source === NetSdkRunTaskType);
+        if (!runTaskFirstEntry) {
+            throw Error(l10n.t('No .NET SDK run task found '));
+        }
+        const taskExecution = await tasks.executeTask(runTaskFirstEntry);
+
+        const taskEndPromise = new Promise<void>((resolve, reject) => {
+            const disposable = tasks.onDidEndTask(e => {
+                if (e.execution === taskExecution) {
+                    disposable.dispose();
+                    resolve();
+                }
+            });
+        });
```

Comment:
I'm nervous about possible timing issues here, since we might never detect the task had ended if it failed immediately. Another argument for executing a custom task is that we can implement our own promise that is resolved by the task itself rather than depending on the event system.

Reply:
The listener could be set up before the call to `vscode.tasks.executeTask`

Reply:
the implementation of provideTasks has been changed by suggestions via @danegsta, this shouldn't be a concern anymore (at least i hope 😆 )